YotiButtonSDK/YotiSDK.swift

```diff
@@ -17,33 +17,48 @@ public class YotiSDK: NSObject {
     @objc public static let willMakeNetworkRequest = Notification.Name("com.yoti.willMakeNetworkRequest")
 
     var kernel = KernelSDK.shared
+
     private var scenarios = [String: Scenario]()
     
+    weak var delegate: YotiSDKDelegate?
+    
     // MARK: - Static Functions
     @objc(addScenario:)
     public static func add(scenario: Scenario) {
         shared.add(scenario: scenario)
     }
     
-    @objc(getScenarioForUseCaseID:)
-    public static func scenario(for useCaseID: String) -> Scenario? {
-        return shared.scenario(for: useCaseID)
+    @objc(startScenarioForUseCaseID:withDelegate:error:)
+    public static func startScenario(for useCaseID: String, with delegate: YotiSDKDelegate) throws {
+        try shared.startScenario(for: useCaseID, with: delegate)
     }
     
-    static func startScenario(for useCaseID: String) throws {
-        try shared.startScenario(for: useCaseID)
+    public static func callbackBackend(scenario: Scenario, token: String, with delegate: BackendDelegate) {
+        shared.callbackBackend(scenario: scenario, token: token, with: delegate)
     }
     
+    public static func scenario(for useCaseID: String) -> Scenario? {
+        return shared.scenario(for:useCaseID)
+    }
+    
+    func callbackBackend(scenario: Scenario, token: String, with delegate: BackendDelegate) {
+        kernel.callbackBackend(scenario: scenario, token: token, with: delegate)
+    }
+
     // MARK: - Instance Functions
     func add(scenario: Scenario) {
         scenarios[scenario.useCaseID] = scenario
     }
     
+    public static func set(delegate: YotiSDKDelegate) {
```

Comment:
The delegate should be scenario specific, a universal delegate is inappropriate.